src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -363,6 +363,15 @@ void
 CreateDistributedTable(Oid relationId, Var *distributionColumn, char distributionMethod,
 					   char *colocateWithTableName, bool viaDeprecatedAPI)
 {
+	List *fKeyCreationCommandsRelationInvolved = NIL;
+	if (IsCitusTableType(relationId, CITUS_LOCAL_TABLE))
+	{
+		/* store foreign key creation commands that relation is involved */
+		fKeyCreationCommandsRelationInvolved =
+			GetFKeyCreationCommandsRelationInvolved(relationId);
+		relationId = DropFKeysAndUndistributeTable(relationId);
```

Comment:
moving this block after `EnsureRelationCanBeDistributed` might make more sense, but we need to do `EnsureRelationCanBeDistributed->EnsureTableNotDistributed` check separately, right after this block

Reply:
this would resolve an important amount of ci failures

Reply:
no, on the contrary, we should definitely undistribute citus local table before those checks as some checks in `EnsureRelationCanBeDistributed` relies on relation is a postgres table.